great_expectations/data_context/store/configuration_store.py

```diff
@@ -0,0 +1,161 @@
+yaml.default_flow_style = False
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigurationStore(Store):
+    """
+    A Configuration Store provides a way to store GreatExpectations Configuration accessible to a Data Context.
+    """
+
+    _key_class = ConfigurationIdentifier
+
+    def __init__(
+        self,
+        configuration_class: BaseConfig,
+        store_name: str,
+        store_backend: dict = None,
+        overwrite_existing: bool = False,
+        runtime_environment: dict = None,
+    ):
+        self._configuration_class = configuration_class
+
+        self._store_name = store_name
+
+        if store_backend is not None:
+            store_backend_module_name = store_backend.get(
+                "module_name", "great_expectations.data_context.store"
+            )
+            store_backend_class_name = store_backend.get(
+                "class_name", "InMemoryStoreBackend"
+            )
+            verify_dynamic_loading_support(module_name=store_backend_module_name)
+            store_backend_class = load_class(
+                store_backend_class_name, store_backend_module_name
+            )
+
+            # Store Backend Class was loaded successfully; verify that it is of a correct subclass.
+            if issubclass(store_backend_class, TupleStoreBackend):
+                # Provide defaults for this common case
+                store_backend["filepath_suffix"] = store_backend.get(
+                    "filepath_suffix", ".yml"
+                )
+
+        super().__init__(
+            store_backend=store_backend, runtime_environment=runtime_environment
+        )
+
+        self._config = get_currently_executing_function_call_arguments(
+            include_module_name=True, **{"class_name": self.__class__.__name__,}
+        )
+        filter_properties_dict(properties=self._config, inplace=True)
+
+        self._overwrite_existing = overwrite_existing
+
+    def load_configuration(self) -> BaseConfig:
+        logger.debug("Starting ConfigurationStore.load_configuration")
+
+        try:
+            key: ConfigurationIdentifier = ConfigurationIdentifier(
+                configuration_name=self.store_name
+            )
+            config_yaml: str = self.get(key)
+            config_commented_map_from_yaml: CommentedMap = yaml.load(config_yaml)
+            try:
+                return self.configuration_class.from_commented_map(
+                    config_commented_map_from_yaml
+                )
+            except ge_exceptions.InvalidBaseConfigError:
+                # Just to be explicit about what we intended to catch
+                raise
+        except ge_exceptions.InvalidKeyError:
+            raise ge_exceptions.ConfigNotFoundError()
+
+    def save_configuration(self, configuration: BaseConfig) -> None:
+        logger.debug("Starting ConfigurationStore.save_configuration")
+
+        # noinspection PyUnusedLocal
+        do_store: bool = False
+        if self.overwrite_existing:
```

Comment:
This could be done the other way -- i.e., by passing it to "save_configuration()".  I opted for making overwrite_existing an instance variable so that the way configuration is saved is intended at the instantiation time, making an errant flip of this flag during usage impossible, thus preventing the unintended overwrite.  Basically, the user has to know their intention behind usage of Configuration as a whole, not at the level of usage of its instance methods.